files/default/slack_handler.rb

```diff
@@ -57,21 +59,51 @@ def report
 
   private
 
-  def run_status_detail
+  def setup_slackr_options(config = {})
+    # options to be passed to slackr gem
+    @slackr_options = {}
+    # icon_url takes precedence over icon_emoji
+    @slackr_options[:icon_url] = config[:icon_url] if config[:icon_url]
+    @slackr_options[:icon_emoji] = config[:icon_emoji] unless config[:icon_url]
+    @slackr_options[:channel] = config[:channel]
+    @slackr_options[:username] = config[:username]
+  end
+
+  def message(context)
+    "Chef client run #{run_status_human_readable} on #{run_status.node.name}#{run_status_cookbook_detail(context['cookbook_detail_level'])}#{run_status_detail(context['detail_level'])}"
+  end
+
+  def run_status_detail(detail_level)
     case detail_level
     when "elapsed"
-      "(#{run_status.elapsed_time} seconds). #{updated_resources.count} resources updated" unless updated_resources.nil?
+      " (#{run_status.elapsed_time} seconds). #{updated_resources.count} resources updated" unless updated_resources.nil?
     when "resources"
-      "(#{run_status.elapsed_time} seconds). #{updated_resources.count} resources updated\n#{updated_resources.join(', ')}" unless updated_resources.nil?
+      " (#{run_status.elapsed_time} seconds). #{updated_resources.count} resources updated\n#{updated_resources.join(', ')}" unless updated_resources.nil?
     end
   end
 
-  def slack_message(content)
-    slack = Slackr.connect(team, api_key, config)
+  def slack_message(content, node_name)
+    slackr_options[:username] = node_name unless slackr_options[:username]
+    slack = Slackr.connect(team, api_key, slackr_options)
     slack.say(content)
   end
 
   def run_status_human_readable
     run_status.success? ? "succeeded" : "failed"
   end
+
+  def run_status_cookbook_detail(detail_level)
+    case detail_level
+    when "off"
+      return
+    when "all"
+      cookbooks = run_status.run_context.cookbook_collection
+      " using cookbooks #{cookbooks.values.map { |x| x.name.to_s + ' ' + x.version }}"
+    when "root"
+      root_cookbook = run_status.run_context.cookbook_collection.values.first
```

Comment:
This does not really represent the "root" cookbook.  It only represents the one that is in the run list.  I think this should be the run_list instead